### ISSUE TITLE
Excessive CPU Usage (%99) when Windows Defender is active

### ISSUE THREAD

**Issue by Frozander** [Gruntfuggly/todo-tree#346]
I don't know if the newest version of VSCode caused this.
While Windows Defender is active, the constantly running rp.exe triggers Windows Defender so it uses all the system resources.

### Steps To Reproduce
- Enable Windows Defender
- Enable Todo Tree addon on VSCode
- Turn on VSCode
- Check Task Manager for CPU Usage

### VSCode Version Info
Version: 1.47.0 (system setup)
Commit: d5e9aa0227e057a60c82568bf31c04730dc15dcd
Date: 2020-07-09T08:02:06.629Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19042

**Comment by Gruntfuggly** [comment]
Hi - could you post your configuration for todo-tree. 

Is your codebase particularly large? If you open the todo-tree debug panel, does it show the search taking a long time?

What's odd is that ripgrep doesn't change any files, so I don't understand why Windows Defender would be triggered?

**Comment by Frozander** [comment]
I didn't change any configuration so it was the default settings.

Codebase is pretty large, I am working on a Machine Learning Project and I had this issue the training data was in the working directory. And that was probably the cause.

However when I first tried to change workspaces, nothing changed. So I did more tests.

Apparently VSCode does not kill rg.exe processes even if it is shutdown when they are doing heavy work like going through 2 gigs of text data. I used "Reload Window" which created even more rg.exe instances (excluding the ones it uses initially for a short time).

When it comes to Windows Defender, I have no idea why it triggers, I am not familiar with how antimalware works. But it seems to only trigger with excessive I/O calls, like reading bunch of tiny text files summed up to gigabytes of data.

In conclusion, I need to use VSCode's files.exclude to exclude my data from my workspace from now on.

**Comment by Gruntfuggly** [comment]
OK thanks for the update. I'll look into killing any search process when code is shut down - they're spawned by the extension so it's probably not VSCode's fault, but mine.

**Comment by Gruntfuggly** [comment]
The ripgrep process should now be killed on shutdown in the latest version.
